content/1_docs/2_cookbook/7_extensions/0_first-panel-view/recipe.txt

```diff
@@ -0,0 +1,358 @@
+
+(screenshot: movie-reviews-custom-view.png)
+
+<info>
+For the purposes of this recipe, we assume that Parcel is already installed globally as described in the (link: docs/guide/plugins/plugin-setup-panel text: Panel plugin setup guide).
+
+You can install the (link: https://github.com/getkirby/pluginkit/tree/4-panel text: pluginkit) as a basis or create the file structure we need manually, that's up to you. Also, it doesn't matter if you use the PlainKit or the StarterKit as a starting point.
+</info>
+
+Let's start by creating a new folder in the plugins folder, which we will call `moviereviews`. Inside this folder, we first create a `package.json` file with the contents copied from the pluginkit example mentioned above.
+
+```json "/site/plugins/moviereviews/package.json"
+{
+  "scripts": {
+    "dev": "parcel watch src/index.js --no-source-maps -d ./",
+    "build": "parcel build src/index.js --no-source-maps --experimental-scope-hoisting -d ./"
+  },
+  "posthtml": {
+    "recognizeSelfClosing": true
+  },
+  "devDependencies": {
+    "@vue/component-compiler-utils": "^3.0.2",
+    "sass": "^1.23.6",
+    "vue-template-compiler": "^2.6.10"
+  },
+  "dependencies": {
+    "vue": "^2.6.10",
+    "vue-hot-reload-api": "^2.3.4"
+  }
+}
+```
+This will take care of compiling our source files into an `index.js` file in the root of our `moviereviews` plugin folder.
+
+
+## Custom API route
+
+Inside the `moviereviews` folder, we create an `index.php` with the Kirby plugin wrapper. And inside this wrapper, we create a custom API route that serves as our data provider. It requests the data from the selected API with a `Remote::get()` request and returns it, so that we can later use it in our component.
+
+For this example, we reuse the (link: https://developer.nytimes.com/docs/movie-reviews-api/1/overview text: NYT Movie Reviews API) we already used in the (link: docs/guide/virtual-pages/content-from-api Virtual Pages guide). To follow this example, create an account and an example app (link: https://developer.nytimes.com/get-started text: by following their instructions). The get your API key and add it to the code below.
+
+```php
+<?php
+
+Kirby::plugin('getkirby/moviereviews', [
+    'api' => [
+        'routes' => [
+            [
+                'pattern' => 'moviereviews',
+                'action'  => function () {
+                    $apikey = 'your api key here';
+                    $request = Remote::request('https://api.nytimes.com/svc/movies/v2/reviews/picks.json?api-key=' . $apikey);
+                    if ($request->code() === 200) {
+                      $reviews = $request->json(false)->results;
+                    }
+                    return $reviews;
+                }
+            ]
+        ]
+    ]
+]);
+```
+
+You can test if the route returns the desired data by calling it in a REST client with Basic Authentication. Make sure to enable `basicAuth` and `allowInsecure` in your config.
```

Comment:
I‘d make a note here that these options should only be set for debugging and removed afterwards unless they are actually required for the specific use-case.

Reply:
✅